### PR TITLE
Use ostringstream to generate id string

### PR DESCRIPTION
**Pull request by jhilmer**
IMHO better to use ostringstream to generate the `getId` string

### CODE REVIEW

File: src/core/gui/toolbarMenubar/ColorToolItem.cpp [hunk 1]
```diff
@@ -70,12 +71,9 @@ auto ColorToolItem::getId() const -> std::string {
...
     }
 
-    // Todo (modernize, cpp20): use std::format or fmtlibs fmt::format
-    std::array<char, 64> buffer{'\0'};
-    auto size = snprintf(buffer.data(), buffer.size(), "COLOR(%zu)", this->namedColor.getIndex());
-    std::string id = {buffer.data(), static_cast<size_t>(size)};
-
-    return id;
+    std::ostringstream os;
+    os << "COLOR(" << this->namedColor.getIndex() << ")" << std::ends;
```
Comment (rolandlo) [inline src/core/gui/toolbarMenubar/ColorToolItem.cpp:75]: Is `std::ends` really needed here?

Comment (jhilmer) [reply src/core/gui/toolbarMenubar/ColorToolItem.cpp:75]: No it's not needed.

**Review by rolandlo — APPROVED** [review]
LGTM

Squashing and merging in 24 hours if no objections are raised.

**Comment by rolandlo** [comment]
Thanks for your contributions, @jhilmer ! Very much appreciated.

**Pull request merged**